### PR TITLE
Re-enable test_copy_transpose_math_view, neg_view/dce fix

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #110770
* __->__ #110651

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

- neg view can just be lowered to neg() post functionalization
- we were treating all fallback kernels as not having side effects. we shouldn't dce mutating fallback kernels - either mutations induced by the reinplacing pass or clone_ with unsupported arguments (complex)

### CODE REVIEW

**Comment by Chillee** [comment]
> either mutations induced by the reinplacing pass 

Wait sorry, when can this result in inplace fallback kernels?

**Review by Chillee — APPROVED** [review]
Oh I see. 🤔

File: torch/_inductor/debug.py [hunk 1]
```diff
@@ -451,6 +451,7 @@ def ir_post_fusion(self, nodes: SchedulerNodeList):
 
     def _write_ir(self, filename: str, nodes: SchedulerNodeList):
         with self.fopen(filename) as fd:
+            print(f"Writing debug ir to {fd.name}")
```
Comment (jansel) [inline torch/_inductor/debug.py:454]: remove or log.info?

**Review by jansel — APPROVED** [review]

File: torch/_inductor/debug.py (hunk 1 above)
Comment (malfet) [inline torch/_inductor/debug.py:454]: Not needed I guess?
```suggestion
```

File: torch/_inductor/ir.py [hunk 2]
```diff
@@ -3846,6 +3847,19 @@ def find_device(tensor_args, example_output):
...
+        # TODO - some fallbacks are still OpOverloadPackets
+        if not isinstance(self.op_overload, torch._ops.OpOverload):
+            return False
+        return get_schema_info(self.op_overload).is_mutable()
+
+    def has_aliasing(self):
+        # TODO - some fallbacks are still OpOverloadPackets
+        if not isinstance(self.op_overload, torch._ops.OpOverload):
+            return False
+
```
Comment (malfet) [inline torch/_inductor/ir.py:3860]: Nit (or add newline between similar calls on line 3853
```suggestion
```

**Review by malfet — APPROVED** [review]

File: torch/_inductor/debug.py (hunk 1 above)
Comment (eellison) [reply torch/_inductor/debug.py:454]: This is only generated if debug logs are already set. Currently we were writing them out but not specifying where they were written so you couldnt find them. I can switch this to using logging though.

Comment (malfet) [reply torch/_inductor/debug.py:454]: Still I think `print` is worse than using a structured logging (@ezyang knows best which mechanism is preferred right now)

**Review by Skylion007 — APPROVED** [review]

**Pull request closed without merging**